src/main/java/duke/Event.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.time.LocalDateTime;
+
+public class Event extends TaskItem {
+
+
+
+    protected boolean readFromFile = false;
+
+    protected LocalDateTime byDateAndTime;
+
+
+    protected String dateAndTimeInString;
+    /**
+     * Constructor for creating an duke.Event object.
+     * @param description description of the event.
+     * @param byDateAndTime date and time of the event.
+     */
+    public Event(String description, LocalDateTime byDateAndTime) {
+        super(description);
+        this.byDateAndTime = byDateAndTime;
+        if (byDateAndTime.getDayOfMonth() == LocalDateTime.now().getDayOfMonth()) {
+            int hour = byDateAndTime.getHour();
+            int minute = byDateAndTime.getMinute();
+            this.dateAndTimeInString = "Today at " + hour + ":" + minute;
+        } else {
+            String timeOfDay;
+            int hour = byDateAndTime.getHour();
+            if (byDateAndTime.getHour() > 12) {
+                timeOfDay = "PM";
+                hour -= 12;
+            } else if (byDateAndTime.getHour() == 12) {
+                timeOfDay = "PM";
+            } else {
+                timeOfDay = "AM";
+            }
+            this.dateAndTimeInString = this.byDateAndTime.getDayOfWeek().toString() + " " + hour + timeOfDay;
```

Comment:
Would you consider formatting the date and time to be stored in the "dateAndTimeInString" by using the methods under the DateTimeFormatter class? This way you can make your code cleaner and print in a specified format too :)